server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -587,6 +595,9 @@ public boolean equals(Object o) {
         if (!inSyncAllocationIds.equals(that.inSyncAllocationIds)) {
             return false;
         }
+        if (!rolloverInfos.equals(that.rolloverInfos)) {
```

Comment:
` == false` (#30697) (I see there are other places in the source that are not like this but they can stay.)